Add barometric altitude to OSD

DanNixon opened the pull request:
Adds the option to display baro altitude on OSD in either metres or feet.

Will need configurator changes which I should get round to in a day or so assuming there is no issue with the proposed changes here.

borisbstyle commented:
Cool. 

This indeed requires configurator changes as well

mikeller commented:
I think this should go into development and not master.

Also, I would consider adding the extra fields to `src/main/serial_msp.c` right away. Stubbing out the data in configurator is simple (and I am happy to do this for you if you're not confident), and once it's in, you can focus on the configurator changes with the data already available.

DanNixon commented:
@mikeller I've changed the base branch now and added the unit selection into MSP (truth be told I had overlooked this until I started working on the configurator).

mikeller commented:
Looking good, can you rebase to squash the commits into one please?
